### ISSUE TITLE
[Bug : Build 24.x] The build fails - 'org.eclipse.equinox.p2.iu; org.hamcrest.core [1.1.0,2.0.0)' could not be found

### ISSUE THREAD

**Issue by wolfgang-ch** [mytourbook/mytourbook#1321]
### What happened ?

The build fails when resolving test dependencies, maybe because of using Eclipse 2024-04

![image](https://github.com/mytourbook/mytourbook/assets/1283445/00a9e1a3-2f01-478d-ab2d-4e0e365c677a)

### Version

24.5

### System

All

**Comment by FJBDev** [comment]
This seems to me that its a duplicate of [this issue](https://github.com/mytourbook/mytourbook/commit/fafdbfe7e92af060068e5a5b029deb32d2979fc8)

**Comment by wolfgang-ch** [comment]
With Tycho 4, this issue disappears but then the p2 issue occurs https://github.com/mytourbook/mytourbook/issues/1176

So I've to fix the p2 issue until Tycho 4 can be used. For now I revert to the old Eclipse version

**Comment by FJBDev** [comment]
> With Tycho 4, this issue disappears but then the p2 issue occurs #1176
> 
> So I've to fix the p2 issue until Tycho 4 can be used. For now I revert to the old Eclipse version

Indeed, Tycho4 fixes the issue. Maybe you should fix the p2 issue, so that we can use the latest Eclipse release for MT 24.5.
But i dont know how hard that is?

**Comment by wolfgang-ch** [comment]
> > With Tycho 4, this issue disappears but then the p2 issue occurs #1176
> > So I've to fix the p2 issue until Tycho 4 can be used. For now I revert to the old Eclipse version
> 
> Indeed, Tycho4 fixes the issue. Maybe you should fix the p2 issue, so that we can use the latest Eclipse release for MT 24.5. But i dont know how hard that is?

MT 24.5 is done for me but for the following version, I will try to fix the p2 issue, I've found some infos how to create it:

* https://www.vogella.com/tutorials/EclipseTycho/article.html#exercise-tycho-build-for-update-sites
* https://stackoverflow.com/questions/66799915/missing-plugin-and-feature-folders-for-an-update-site-built-with-maven-tycho

**Comment by FJBDev** [comment]
> MT 24.5 is done for me but for the following version, I will try to fix the p2 issue, I've found some infos how to create a p2 update site:

FYI, this is my [PR ](https://github.com/mytourbook/mytourbook/pull/1192 )in draft mode from 6 months ago to use tycho 4 and the latest Eclipse release.
Once you fix the p2 issue, I can activate the PR again to merge it in `main`

**Comment by FJBDev** [comment]
Similar issue [here](https://github.com/de-jcup/eclipse-commons/issues/55) that someone did a workaround for it

**Comment by wolfgang-ch** [comment]
After 3 days of intense try and errors, I give up for now for using a tycho repo for the eclipse nl plugins https://github.com/mytourbook/mytourbook/commit/ca63c1098a01dd010d09989d168fd4e44023b7fb

What I've learned:
* In tycho error messages, e.g.
`[ERROR]   Missing requirement: feature.net.tourbook.ext.eclipse.nl.feature.group 4.23.0.qualifier requires 'org.eclipse.equinox.p2.iu; org.eclipse.core.commands.nl_de 0.0.0' but it could not be found`
the name `org.eclipse.equinox.p2.iu` is just a namespace and NOT any name of a plugin, the real issue is the name afterwards, including the version number

* The problem with the p2 tycho repo may be, that the eclipse nl plugins are plugin fragments without any java code and they were not put into the p2 repo in my tests
https://www.google.com/search?q=%22fragment-host%22+%22tycho%22+update+site

**Comment by FJBDev** [comment]
@wolfgang-ch 

Did you try to launch the build without the test modules references [here ](https://github.com/mytourbook/mytourbook/blob/a43e131f1b858e3e1176f5d38065a4441f3da0d5/bundles/pom.xml#L40)? Since they are only needed for the Github build (they are skipped by default [here](https://github.com/mytourbook/mytourbook/blob/a43e131f1b858e3e1176f5d38065a4441f3da0d5/pom.xml#L177)), I am curious to see if running the build on your local machine without those test modules references would fix that issue.

If it does, then I could tweak the build to add those reference ONLY in the Github build so that we can continue updating MT with the recent Eclipse versions.
let me know

**Comment by wolfgang-ch** [comment]
The problem is Tycho 4, building MT including the tests with Tycho 3 and Eclipse 2406 is OK

![image](https://github.com/user-attachments/assets/6144c48f-c573-49d1-9c2c-dc340229079d)


[mt24.8-with-eclipse-2406-log.txt](https://github.com/user-attachments/files/16556992/mt24.8-with-eclipse-2406-log.txt)

**Comment by FJBDev** [comment]
> The problem is Tycho 4, building MT including the tests with Tycho 3 and Eclipse 2406 is OK
> 

I am confused then.... Why not using Eclipse 202406 in the [MT 24.8](https://github.com/mytourbook/mytourbook/blob/main/pom.xml#L76) if "Tycho 3 and Eclipse 2406 is OK" ??

**Comment by wolfgang-ch** [comment]
> > The problem is Tycho 4, building MT including the tests with Tycho 3 and Eclipse 2406 is OK
> 
> I am confused then.... Why not using Eclipse 202406 in the [MT 24.8](https://github.com/mytourbook/mytourbook/blob/main/pom.xml#L76) if "Tycho 3 and Eclipse 2406 is OK" ??

We could have used Eclipse 2406 but for me there was no need

**Comment by wolfgang-ch** [comment]
@FJBDev Yesterday I tested the build process with Eclipse 4.33 and this issue still happens, so we need to fix this anywhere to use newer Eclipse versions

**Comment by FJBDev** [comment]
> @FJBDev Yesterday I tested the build process with Eclipse 4.33 and this issue still happens, so we need to fix this anywhere to use newer Eclipse versions

Let's update to 202406 first since the issue doesn't happen with 2024-06/4.33.

However, I tried to launch an MT build with 2024-09 and it works fine in GitHub

![image](https://github.com/user-attachments/assets/1e74c32a-d94b-4e16-ba6e-416378d720b5)

**Comment by FJBDev** [comment]
@wolfgang-ch Did you try to update to Tycho 4 ? It seems to work with Tycho 4

`<tycho.version>4.0.8</tycho.version>`

**Comment by FJBDev** [comment]
This [thread ](https://github.com/de-jcup/eclipse-commons/issues/55)seems to contain a local build fix but i dint try it locally
